unpackaged/vesuvio_calibration/calibrate_vesuvio_uranium_martyn_MK5.py

```diff
@@ -379,114 +379,155 @@ def _normalise_sample_by_background(self):
+                                          workspace_index, fit_peaks_output_name, x_range)
+      fit_results['status'] = "peaks invalid" if not \
+          self._check_fitted_peak_validity(fit_peaks_output_name + '_Parameters', peak_estimates_list,
+                                           peak_height_rel_threshold=PEAK_HEIGHT_RELATIVE_THRESHOLD) else fit_results[
+                                           'status']
+      return fit_results
+
+  def _run_find_peaks(self, workspace_index, find_peaks_output_name, find_peaks_input_params, unconstrained):
+      try:
+          FindPeaks(InputWorkspace=self._sample, WorkspaceIndex=workspace_index, PeaksList=find_peaks_output_name,
+                    **find_peaks_input_params)
+          if mtd[find_peaks_output_name].rowCount() > 0:
+              peaks_found = True
+          else:
+              raise ValueError
+      except ValueError:
+          peaks_found = False
+          if not unconstrained: #Ignore error if unconstrained, as we will use peaks found during constrained workflow.
+              raise ValueError("Error finding peaks.")
+      return peaks_found
+
+  def _select_best_fit_params(self, spec_num, fit_results, fit_results_u=None):
+      selected_params = fit_results['params']
+      unconstrained_fit_selected = False
+      if fit_results_u:
+          if fit_results_u['chi2'] < fit_results['chi2'] and fit_results_u['status'] != "peaks invalid":
+              selected_params = fit_results_u['params']
+              unconstrained_fit_selected = True
+              self._prog_reporter.report("Fit to spectrum %d without constraining parameters successful" % spec_num)
+      return selected_params, unconstrained_fit_selected
+
+  def _output_params_to_table(self, spec_num, num_estimated_peaks, params, output_table_name):
+      fit_values = dict(zip(params.column(0), params.column(1)))
+      fit_errors = dict(zip(params.column(0), params.column(2)))
+
+      row_values = [fit_values['f0.A0'], fit_values['f0.A1']]
+      row_errors = [fit_errors['f0.A0'], fit_errors['f0.A1']]
+      param_names = self._get_param_names(num_estimated_peaks)
+      row_values += [fit_values['f1.' + name] for name in param_names[2:]]
+      row_errors += [fit_errors['f1.' + name] for name in param_names[2:]]
+      row = [element for tupl in zip(row_values, row_errors) for element in tupl]
+
+      mtd[output_table_name].addRow([spec_num] + row)
+
+  def _get_output_and_clean_workspaces(self, unconstrained_fit_performed, unconstrained_fit_selected, find_peaks_output_name,
+                                       fit_peaks_output_name):
+      find_peaks_output_name_u = self._get_unconstrained_ws_name(find_peaks_output_name)
+      fit_peaks_output_name_u = self._get_unconstrained_ws_name(fit_peaks_output_name)
+
+      DeleteWorkspace(fit_peaks_output_name + '_NormalisedCovarianceMatrix')
```

Comment:
The `_Workspace` and other suffixes seemed to be used multiple times here. Would it be possible to make it a named constant at the top of the file e.g. `WORKSPACE_SUFFIX = "_Workspace"`?

Reply:
I think it would be good to rationalise workspace naming/the use of suffixes throughout the whole script, as it's quite hard to follow. I'd prefer to do this at once in a subsequent PR though.